src/main/webapp/script.js

```diff
@@ -276,63 +276,72 @@ class AutocompleteDirectionsHandler {
 
             for (const step of leg["steps"]) {
                 const stepIndex = leg["steps"].indexOf(step);
-                console.log("Printing out start point of a step in a leg. In step " + (stepIndex + 1) + " of " + leg["steps"].length + step["start_location"]);
+                console.log("Printing out start point of a step in a leg. In step " + (stepIndex + 1) + " of " + leg["steps"].length);
 
                 // The duration value indicates duration in seconds. Using that (time) as stepWeight. 
                 let stepWeight = step["duration"]["value"];
                 let stepStartLat = step["start_location"].lat();
                 let stepStartLng = step["start_location"].lng();
+                console.log("step coord" + step["start_location"] + " time weight: " + stepWeight);
 
                 let mapRowCol = getGridIndex(stepStartLat, stepStartLng, aqDataPointsPerDegree);
                 let mapRow = mapRowCol.row;
                 let mapCol = mapRowCol.col;
 
                 // AQI is only available for routes passing through the grid exactly. 
                 // Or else no data available and will have error saying it is undefined (because it doesnt exist). 
-                // If the mapRow from getGridIndex of stepAQIdoesnt doesn't exist in the data grid, stepAQI is 0 for now. 
-                let stepAQI = 0;
+                // Ignore step if the mapRow from getGridIndex of stepAQIdoesnt doesn't exist in the data grid, 
+                // as setting it to 0 will skew data to 0 if passing through an area with very little data. 
+                // So the route score will only be scored based on available data. 
+                var stepAQI;
```

Comment:
It's best not to use `var`, it can cause weird behaviour sometimes. You can replace it with `let`

Reply:
going to clean up console log comments so they're more succint.